src/main/java/org/codehaus/mojo/versions/AbstractVersionsReport.java

```diff
@@ -234,18 +234,10 @@ protected abstract void doGenerateReport( Locale locale, Sink sink )
      * @since 1.0-alpha-1
      */
     protected ArtifactVersion findLatestVersion( Artifact artifact, VersionRange versionRange,
-                                                 boolean allowingSnapshots, boolean usePluginRepositories )
+                                                 Boolean allowingSnapshots, boolean usePluginRepositories )
         throws MavenReportException
     {
-        boolean includeSnapshots = this.allowSnapshots;
-        if ( allowingSnapshots )
-        {
-            includeSnapshots = true;
-        }
-        if ( allowingSnapshots )
-        {
-            includeSnapshots = false;
-        }
```

Comment:
Interesting ... looks like never was working

Reply:
No, not really. I contemplated removing it too, but I saw it being used.

this.allowSnapshots was a `Boolean` value, could also be null. Local `allowingSnaphots` was a primitive `boolean`, but it was a kind of an override for the value in the Mojo.

So, if `allowingSnapshots` method argument was set, the code overrote the value from the Mojo object with the method argument.

Reply:
There are two methods with Boolean overrides. One of them was with the boolean primitive and seems not used. I did not felt removing this one. Might consider marking deprecated ?